### PR TITLE
Fixed #3114 - Page navigator does not change state when scrolling

### LINKED ISSUE #3114: Page navigator does not change state when scrolling
**We need test for it!**
image

### PR DESCRIPTION
**Pull request by tsv2013**

### CODE REVIEW

File: packages/survey-creator-knockout/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
         "doc:gen": "node doc_generator/editor_docgenerator.js src/entries/index.ts",
         "doc:update": "chmod +x docupdate_npm.sh && ./docupdate_npm.sh",
         "testcafe": "concurrently \"http-server --silent\" \"testcafe -c 4 chrome ../../testCafe/ --selector-timeout 1500\"",
-        "testcafe:file": "concurrently \"http-server --silent\" \"testcafe chrome ../../testCafe/designer/page-edit-modes.ts --reporter minimal --selector-timeout 1500 --debug-mode\"",
+        "testcafe:file": "concurrently \"http-server --silent\" \"testcafe chrome ../../testCafe/designer/page-navigator.ts --reporter minimal --selector-timeout 1500 --debug-mode\"",
         "testcafe:a11y": "concurrently \"http-server --silent\" \"testcafe chrome ./testCafe/accessibility.js --selector-timeout 1500\"",
```
Comment (OlgaLarina) [inline packages/survey-creator-knockout/package.json:24]: Maybe you should not give changes to package.json

File: testCafe/designer/page-navigator.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+      ]
+    };
+  })();
+
+  await t
+    .expect(Selector("svc-page-navigator-item:nth-child(2) > .svc-page-navigator-item--selected, .svc-page-navigator-item:nth-child(2) > .svc-page-navigator-item--selected").exists).ok()
+    .scroll(Selector(".svc-tab-designer--with-page-navigator"), "bottomRight")
+    .scroll(Selector(".svc-tab-designer--with-page-navigator"), "bottomRight")
+    .expect(Selector("svc-page-navigator-item:nth-child(5) > .svc-page-navigator-item--selected, .svc-page-navigator-item:nth-child(5) > .svc-page-navigator-item--selected").exists).ok();
```
Comment (OlgaLarina) [inline testCafe/designer/page-navigator.ts:58]: It is not clear what is being tested in the test

File: packages/survey-creator-react/package.json [hunk 3]
```diff
@@ -21,7 +21,7 @@
     "doc:update": "chmod +x docupdate_npm.sh && ./docupdate_npm.sh",
     "testcafe": "concurrently \"http-server --silent\" \"testcafe -c 4 chrome ../../testCafe/ --selector-timeout 1500\"",
     "testcafe:ci": "http-server --silent & testcafe -c 4 -q chrome:headless ../../testCafe/ --selector-timeout 1500 --reporter minimal,dashboard",
-    "testcafe:file": "concurrently \"http-server --silent\" \"testcafe chrome ../../testCafe/property-grid/test.ts --reporter minimal --debug-mode\"",
+    "testcafe:file": "concurrently \"http-server --silent\" \"testcafe chrome ../../testCafe/designer/page-navigator.ts --reporter minimal --debug-mode\"",
```
Comment (OlgaLarina) [inline packages/survey-creator-react/package.json:24]: Maybe you should not give changes to package.json

**Review by OlgaLarina — CHANGES_REQUESTED** [review]

File: testCafe/designer/page-navigator.ts (hunk 2 above)
Comment (tsv2013) [reply testCafe/designer/page-navigator.ts:58]: As written in bug title "Page navigator does not change state when scrolling" the test checks exactly this: scrolls to bottom and check last page marker is active

File: packages/survey-creator-knockout/package.json (hunk 1 above)
Comment (tsv2013) [reply packages/survey-creator-knockout/package.json:24]: Why? It doesn't matter.

File: packages/survey-creator-react/package.json (hunk 3 above)
Comment (tsv2013) [reply packages/survey-creator-react/package.json:24]: Why? It doesn't matter.

File: testCafe/designer/page-navigator.ts (hunk 2 above)
Comment (OlgaLarina) [reply testCafe/designer/page-navigator.ts:58]: It is not clear from these long selectors which marker is which. Maybe they can at least take them out and check that the state changes for everyone, after scrolling.

**Review by OlgaLarina — CHANGES_REQUESTED** [review]

**Pull request merged**